scheduler/src/cook/mesos/api.clj

```diff
@@ -71,26 +75,32 @@
 (def NonEmptyString
   (s/both s/Str (s/pred #(not (zero? (count %))) 'empty-string)))
 
-(def Job
-  "A schema for a job"
-  {:uuid (s/pred #(instance? UUID %) 'uuid?)
-   :command s/Str
-   ;; Make sure the job name is a valid string which can only contain '.', '_', '-' or any word characters and has
-   ;; length at most 128.
-   :name (s/both s/Str (s/pred #(re-matches #"[\.a-zA-Z0-9_-]{0,128}" %) 'under-128-characters-and-alphanum?))
-   :priority (s/both s/Int (s/pred #(<= 0 % 100) 'between-0-and-100))
-   :max-retries (s/both s/Int (s/pred pos? 'pos?))
-   :max-runtime (s/both s/Int (s/pred pos? 'pos?))
+
+(s/defrecord Job
+    [uuid :- (s/pred #(instance? UUID %) 'uuid?)
+     command :- s/Str
+     ;; Make sure the job name is a valid string which can only contain '.', '_', '-' or any word characters and has
+     ;; length at most 128.
+     name :- (s/both s/Str (s/pred #(re-matches #"[\.a-zA-Z0-9_-]{0,128}" %) 'under-128-characters-and-alphanum?))
+     priority :- (s/both s/Int (s/pred #(<= 0 % 100) 'between-0-and-100))
+     max-retries :- (s/both s/Int (s/pred pos? 'pos?))
+     max-runtime :- (s/both s/Int (s/pred pos? 'pos?))
+     cpus :- PosDouble
+     mem :- PosDouble
+     ;; Make sure the user name is valid. It must begin with a lower case character, end with
+     ;; a lower case character or a digit, and has length between 2 to (62 + 2).
+     user :- (s/both s/Str (s/pred #(re-matches #"\A[a-z][a-z0-9_-]{0,62}[a-z0-9]\z" %) 'lowercase-alphanum?))     
```

Comment:
I don't know. That requirement was in the pre-existing schema, which I just converted into an s/defrecord.

I'd guess it's probably a Mesos-level constraint, though I haven't been able to find anything specific about it after a quick look at the Mesos docs.